src/daemon/Display.cpp

```diff
@@ -333,26 +356,41 @@ namespace SDDM {
                 emit loginSucceeded(m_socket);
         } else if (m_socket) {
             qDebug() << "Authentication failure";
-            emit loginFailed(m_socket);
+            // avoid to emit loginFailed twice
+            if(!m_failed) {
+                emit loginFailed(m_socket, QString());
+                m_failed = true;
+            }
         }
         m_socket = nullptr;
     }
 
+    // presentable to the user
     void Display::slotAuthInfo(const QString &message, Auth::Info info) {
-        // TODO: presentable to the user, eventually
-        Q_UNUSED(info);
+        if(info == Auth::INFO_PASS_CHANGE_REQUIRED ||
+           info == Auth::INFO_PAM_CONV)
+        {
+            // send pam conversation message to greeter
+            emit pamConvMsg(m_socket, message);
```

Comment:
Probably should check if m_socket is valid here, I've been having a crash with this stacktrace : 
`#0  0x00007f7b9afdcc3d in QIODevice::write(char const*, long long) ()
   from /usr/lib64/libQt5Core.so.5
#1  0x00000000004323ed in SDDM::SocketWriter::~SocketWriter() ()
#2  0x00000000004509fe in SDDM::SocketServer::pamConvMsg(QLocalSocket*, QString const&) ()
#3  0x00007f7b9b0b5831 in QMetaObject::activate(QObject*, int, int, void**) ()
   from /usr/lib64/libQt5Core.so.5
#4  0x0000000000455547 in SDDM::Display::pamConvMsg(QLocalSocket*, QString const&) ()
#5  0x000000000043d497 in SDDM::Display::slotAuthInfo(QString const&, SDDM::Auth::Info) ()
#6  0x00007f7b9b0b5831 in QMetaObject::activate(QObject*, int, int, void**) ()
   from /usr/lib64/libQt5Core.so.5
#7  0x0000000000455386 in SDDM::Auth::info(QString, SDDM::Auth::Info) ()
#8  0x0000000000434724 in SDDM::Auth::Private::dataPending() ()
`

It's caused by pam sending "creating directory" conv msg after auth success at first login

Reply:
hi, the changes for that were already laying around for some time, just had to transfer them to git. is done in the next 3 commits. just have to test them in my new build.